lib/options.nix

```
@@ -275,16 +275,16 @@ rec {
     if ! isString text then throw "literalExpression expects a string."
     else { _type = "literalExpression"; inherit text; };
 
-  literalExample = lib.warn "literalExample is deprecated, use literalExpression instead, or use literalDocBook for a non-Nix description." literalExpression;
-
-
   /* For use in the `defaultText` and `example` option attributes. Causes the
      given DocBook text to be inserted verbatim in the documentation, for when
      a `literalExpression` would be too hard to read.
   */
   literalDocBook = text:
     if ! isString text then throw "literalDocBook expects a string."
-    else { _type = "literalDocBook"; inherit text; };
+    else
+      lib.warn
+        "literalDocBook is deprecated, use literalMD instead"
+        { _type = "literalDocBook"; inherit text; };
```

This will annoy third party module maintainers.
They'll receive issues and PRs about the warnings, while they need to keep using docbook until 22.05 is EOL.
You could prepare the deprecation though, with

```suggestion
    else
      lib.warnIf (lib.isInOldestRelease 2211)
        "literalDocBook is deprecated, use literalMD instead"
        { _type = "literalDocBook"; inherit text; };
```

---

They can't switch because they need to keep using docbook until 22.05 is EOL.

Delaying the warnings until the moment they can switch solves the problem for them.

For reference, this problem was described in https://github.com/NixOS/nixpkgs/issues/157667.
Machinery to automate deprecations was added in https://github.com/NixOS/nixpkgs/pull/163451.